src/coordsim/simulation/simulatorparams.py

```diff
@@ -58,6 +57,31 @@ def __init__(self, network, ing_nodes, sfc_list, sf_list, config, seed, schedule
         # The duration of a run in the simulator's interface
         self.run_duration = config['run_duration']
 
+        self.use_states = False
+        self.states = {}
+        self.in_init_state = True
+
+        if 'use_states' in config and config['use_states']:
+            self.use_states = True
+            self.init_state = config['init_state']
+            self.states = config['states']
+            if self.in_init_state:
+                self.current_state = self.init_state
+            self.inter_arr_mean = self.states[self.current_state]['inter_arr_mean']
+
+    def update_state(self):
+        switch = [False, True]
+        change_prob = self.states[self.current_state]['switch_p']
+        remain_prob = 1 - change_prob
+        switch_decision = np.random.choice(switch, p=[remain_prob, change_prob])
```

Comment:
Minor: Why not directly put `[False, True]` there if you don't need `switch` elsewhere?